### ISSUE TITLE
Force to unzip all wheels from pex

### ISSUE THREAD

**Issue by shvilime** [pantsbuild/pex#2208]
Is it possible to force unzip all wheels from pex ?
I've catch the situation where one of py file from wheel was corrupted and my application wasn't start.
Anyway the pex was sure that the whole wheels were unzipped and no need to repair the problem.
Only when I deleted /user/.pex folder, the pex unzipped again and can start.

**Comment by jsirois** [comment]
No, what you did is basically ^1 the only way. It is assumed the `~/.pex` cache is never tampered with since checking the hash of unzipped files to detect corruption before each launch would add startup overhead that many users would find unacceptable.

Do you have any idea how `~/.pex` became corrupted?

^1: You could say `PEX_ROOT=$(mktemp -d) ./my.pex` to force a new unzip, but that would only work for that run.

**Comment by shvilime** [comment]
I think, in that case the python file was corrupted, due to the work internal utility that inspect and clear old unused temporary files on our servers. This is our internal problem, of course.
Many thanks for your advise, I'll try to fit it to use. Anyway, for me it would be perfect to have a runtime pex param, like --force-unzip, to run it periodically on cron to be sure that all wheels are unpacked and installed again.

**Comment by jsirois** [comment]
> Anyway, for me it would be perfect to have a runtime pex param, like --force-unzip, to run it periodically on cron to be sure that all wheels are unpacked and installed again.

That needs no tooling in Pex. The following is a safe way to kill the entire Pex cache periodically regardless of what other concurrent PEX process happen to be running at the time:
```bash
# Create a directory to atomically move the PEX_ROOT to that lives on the same filesystem.
safe_to_delete_dir="$(mktemp -d -p "${PEX_ROOT}")"

mv "${PEX_ROOT}" "${safe_to_delete_dir}"
rm -rf "${safe_to_delete_dir}"
```

You could cron that script.

**Comment by jsirois** [comment]
@shvilime I'm going to mark this issue as closed since the feature you want is provided by standard POSIX tools and guarantees.
